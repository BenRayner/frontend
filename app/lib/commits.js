export function shortMessage(message) {
  return message.split("\n")[0];
}

export function shortCommit(commitish) {
  // Does this look like a git sha?
  if (commitish && commitish.length === 40 && !commitish.match(/[^a-f0-9]/i)) {
    return commitish.substring(0, 7);
  } else {
    return commitish;
  }
}
