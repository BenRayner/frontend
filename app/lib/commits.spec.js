/* global describe, test, expect */
import { shortMessage, shortCommit } from './commits';

describe('shortMessage', () => {
  const BIG_MESSAGE = `âœ¨ Somehow this is the first spec I've written for our front-end, and this seems really odd to me

- this should be a big enough message to make sure we're not running into anything silly`;

  test('returns the first line of a multi-line message', () => {
    expect(shortMessage(BIG_MESSAGE)).toMatchSnapshot();
  });

  test('returns the whole message in a one-line message', () => {
    expect(shortMessage('Bump frontend')).toMatchSnapshot();
  });
});

describe('shortCommit', () => {
  test('returns a 7-character SHA1 for a full SHA1 commitish', () => {
    expect(shortCommit('be9c03b0dbddafccb0882f62182b84924579c93b')).toMatchSnapshot();
  });

  test('returns the full name for a non-SHA1 commitish', () => {
    expect(shortCommit('preload-pipeline-names-sperlunking')).toMatchSnapshot();
  });

  test('returns the full name for a 40-character non-SHA1 commitish', () => {
    expect(shortCommit('so-this-is-a-forty-character-branch-name')).toMatchSnapshot();
  });
});
