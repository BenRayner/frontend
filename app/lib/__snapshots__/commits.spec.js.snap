exports[`shortCommit returns a 7-character SHA1 for a full SHA1 commitish 1`] = `"be9c03b"`;

exports[`shortCommit returns the full name for a 40-character non-SHA1 commitish 1`] = `"so-this-is-a-forty-character-branch-name"`;

exports[`shortCommit returns the full name for a non-SHA1 commitish 1`] = `"preload-pipeline-names-sperlunking"`;

exports[`shortMessage returns the first line of a multi-line message 1`] = `"âœ¨ Somehow this is the first spec I\'ve written for our front-end, and this seems really odd to me"`;

exports[`shortMessage returns the whole message in a one-line message 1`] = `"Bump frontend"`;
